### PR TITLE
Major update of API documentation

### PR DESCRIPTION
**Pull request by mbtools**
Filled in many details and closed todos. 
Closes https://github.com/abapGit/abapGit/issues/3926

### CODE REVIEW

File: docs/collections/_development/api.md [hunk 1]
```diff
@@ -5,9 +5,13 @@ order: 80
 
 *******************************
 
-## Repositories
+This page describes how to execute various abapGit tasks using your own code. These classes and methods have exists for quite some time and are stable. However, they do **not** provide a guaranteed API. Future change are a possibility.
```
Comment (larshp) [inline docs/collections/_development/api.md:8]:
```suggestion
This page describes how to execute various abapGit tasks using your own code. These classes and methods have existed for quite some time and are stable. However, they do **not** provide a guaranteed API. Future change are a possibility. 
```

File: docs/collections/_development/api.md [hunk 2]
```diff
@@ -5,9 +5,13 @@ order: 80
 
 *******************************
 
-## Repositories
+This page describes how to execute various abapGit tasks using your own code. These classes and methods have existed for quite some time and are stable. However, they do **not** provide a guaranteed API. Future change are a possibility.
```
Comment (FreHu) [inline docs/collections/_development/api.md:8]:
```suggestion
This page describes how to execute various abapGit tasks using your own code. These classes and methods have existed for quite some time and are stable. However, they do **not** provide a guaranteed API. Future changes are a possibility. 
```

File: docs/collections/_development/api.md [hunk 3]
```diff
@@ -16,91 +20,175 @@ DATA(lo_repo) = zcl_abapgit_repo_srv=>get_instance( )->new_online(
...
-## Branches
+## Online Repository ##
 
-### List
-`zcl_abapgit_git_transport=>branches( io_repo->get_url( ) ).`
+The following tasks are supported for online repositories, only (`lo_repo type ref to zcl_abapgit_repo_online`).
 
-### Switching
-`repo->set_branch_name( lv_name )`
+**Note:** Certain tasks will require authentification (user/password or token). In such cases, you will have to provide the login details upfront (see [#1331](https://github.com/abapGit/abapGit/issues/1331) for details):
```
Comment (FreHu) [inline docs/collections/_development/api.md:111]:
```suggestion
**Note:** Certain tasks will require authentication (user/password or token). In such cases, you will have to provide the login details upfront (see [#1331](https://github.com/abapGit/abapGit/issues/1331) for details):
```

File: docs/collections/_development/api.md [hunk 4]
```diff
@@ -16,91 +20,175 @@ DATA(lo_repo) = zcl_abapgit_repo_srv=>get_instance( )->new_online(
...
+```abap
+zcl_abapgit_git_transport=>branches( lo_repo->get_url( ) ).
+```
 
-Online projects, todo, see
-[https://github.com/abapGit/abapGit/blob/master/src/zcl_abapgit_background.clas.abap](https://github.com/abapGit/abapGit/blob/master/src/zcl_abapgit_background.clas.abap)
+### Switch Branch ###
 
-todo: staging, commits, tags
+Change abapGit repository to a different branch:
```
Comment (FreHu) [inline docs/collections/_development/api.md:130]:
```suggestion
Switch abapGit repository to a different branch:
```

File: docs/collections/_development/api.md [hunk 5]
```diff
@@ -16,91 +20,175 @@ DATA(lo_repo) = zcl_abapgit_repo_srv=>get_instance( )->new_online(
...
+Create a new branch in the online repository:
+
+```abap
+lo_repo->create_branch( lv_name ).
+```
 
-## Pulling, Online projects
+### Delete Branch ###
+
+Delete a branch of the online repository:
```
Comment (FreHu) [inline docs/collections/_development/api.md:146]:
```suggestion
Delete a branch of an online repository:
```

File: docs/collections/_development/api.md [hunk 6]
```diff
@@ -16,91 +20,175 @@ DATA(lo_repo) = zcl_abapgit_repo_srv=>get_instance( )->new_online(
...
+Change abapGit repository to a different branch:
 
-*******************************
+```abap
+lo_repo->set_branch_name( lv_name ).
+```
+
+### Create Branch ###
+
+Create a new branch in the online repository:
```
Comment (FreHu) [inline docs/collections/_development/api.md:138]:
```suggestion
Create a new branch in an online repository:
```

File: docs/collections/_development/api.md [hunk 7]
```diff
@@ -143,17 +231,43 @@ METHOD decisions.
...
 ENDMETHOD.
 ```
 
-todo:
-[https://github.com/abapGit/abapGit/issues/1331](https://github.com/abapGit/abapGit/issues/1331)
+*******************************
+
+## Offline Repository ##
+
+The following tasks are supported for offline repositories, only (`lo_repo type ref to zcl_abapgit_repo_offline`).
```
Comment (FreHu) [inline docs/collections/_development/api.md:244]:
```suggestion
The following tasks are supported for offline repositories only (`lo_repo type ref to zcl_abapgit_repo_offline`).
```

File: docs/collections/_development/api.md [hunk 8]
```diff
@@ -143,17 +231,43 @@ METHOD decisions.
...
+
+### Import ZIP ###
+
+Upload ZIP file from frontend to offline repository:
```
Comment (FreHu) [inline docs/collections/_development/api.md:248]:
```suggestion
Upload ZIP file from frontend to an offline repository:
```

File: docs/collections/_development/api.md [hunk 9]
```diff
@@ -143,17 +231,43 @@ METHOD decisions.
...
+```abap
+lv_xstr = zcl_abapgit_ui_factory=>get_frontend_services( )->file_upload( lv_file_with_path ).
+
+lo_repo->set_files_remote( zcl_abapgit_zip=>load( lv_xstr ) ).
+zcl_abapgit_services_repo=>refresh( lv_key ).
+```
+
+### Export ZIP ###
+
+Download ZIP file of offline repository to frontend:
```
Comment (FreHu) [inline docs/collections/_development/api.md:259]:
```suggestion
Download ZIP file of an offline repository to frontend:
```

**Review by FreHu — COMMENTED** [review]
just some grammar and consistency

File: docs/collections/_development/api.md (hunk 3 above)
Comment (mbtools) [reply docs/collections/_development/api.md]: done

**Comment by larshp** [comment]
perhaps my other anchor workaround can be removed, if this works wrt anchors

**Pull request merged**